### PR TITLE
Simplify tautologies

### PR DESCRIPTION
**Pull request by haz**
Idea is to allow for tautologies in `to_CNF` conversions so that model counts aren't thrown off (variables mentioned in a tautological clause will be missing from the compilation).

### CODE REVIEW

**Comment by haz** [comment]
Closes #30

**Review by blyxxyz — APPROVED** [review]
LGTM.

Another solution might be to give `.model_count()` a `names` argument so it can adjust for excluded names. But this is simpler and more broadly useful.

You could add a hypothesis test case that runs `.to_CNF(simplify_tautologies=False)` and checks that no names were lost.

**Comment by haz** [comment]
Great idea! It caught a corner case where `(x & ~x) | y` was being simplified to just `y`. Can you have another look after the latest commit?

**Comment by blyxxyz** [comment]
Ah! I hadn't spotted that.

My only concern now is that the name `simplify_tautologies` is inaccurate because it also covers contradictions.

**Comment by haz** [comment]
Ah, right. Just `simplify` work?

File: nnf/tseitin.py [hunk 1]
```diff
@@ -10,10 +10,11 @@
 from nnf.util import memoize
 
 
-def to_CNF(theory: NNF) -> And[Or[Var]]:
+def to_CNF(theory: NNF, simplify: bool = True) -> And[Or[Var]]:
     """Convert an NNF into CNF using the Tseitin Encoding.
 
     :param theory: Theory to convert.
+    :param simplify: If True, remove clauses that are always true.
```
Comment (blyxxyz) [inline nnf/tseitin.py:17]:
```suggestion
    :param simplify: If True, simplify clauses even if that means eliminating
                     variables.
```
This clarifies why you'd want it to be `False`.

**Review by blyxxyz — CHANGES_REQUESTED** [review]

**Comment by haz** [comment]
Thanks, @blyxxyz ! Should be all set to go if you want to approve/merge.

**Review by blyxxyz — APPROVED** [review]
Looks good! Thanks for bearing with me.
